src/directConnect.ts

```diff
@@ -31,53 +33,138 @@ export function openDirectConnectionForm(): void {
+    if (body.dryRun) result = parseTestResult(connection);
+    else {
+      // save and close the form
+      let name = body.data["name"] || "the connection";
+      if (result.success) {
+        await window.showInformationMessage(`🎉 New Connection Created`, {
+          modal: true,
+          detail: `View and interact with ${name} in the Resources sidebar`,
+        });
+        directConnectForm.dispose();
+      }
     }
     return result;
   }
 
   const processMessage = async (...[type, body]: Parameters<MessageSender>) => {
     switch (type) {
-      case "TestConnection":
-        return (await testConnect(body)) satisfies MessageResponse<"TestConnection">;
       case "Submit":
-        return (await createConnection(body)) satisfies MessageResponse<"Submit">;
+        return (await testOrSaveConnection(body)) satisfies MessageResponse<"Submit">;
     }
   };
   const disposable = handleWebviewMessage(directConnectForm.webview, processMessage);
   directConnectForm.onDidDispose(() => disposable.dispose());
 }
+
+export function getConnectionSpecFromFormData(formData: any): CustomConnectionSpec {
+  const connectionId = randomUUID() as ConnectionId;
+  const spec: CustomConnectionSpec = {
+    id: connectionId,
+    name: formData["name"] || "New Connection",
+    type: ConnectionType.Direct,
+    formConnectionType: formData["platform"],
+  };
+
+  if (formData["bootstrap_servers"]) {
+    spec.kafka_cluster = {
+      bootstrap_servers: formData["bootstrap_servers"],
+    };
+    if (formData.kafka_auth_type === "Basic") {
+      spec.kafka_cluster.credentials = {
+        username: formData["kafka_username"],
+        password: formData["kafka_password"],
+      };
+    } else if (formData.kafka_auth_type === "API") {
+      spec.kafka_cluster.credentials = {
+        api_key: formData["kafka_api_key"],
+        api_secret: formData["kafka_api_secret"],
+      };
+    }
+  }
+
+  if (formData["uri"]) {
+    spec.schema_registry = {
+      uri: formData["uri"],
+    };
+    if (formData.schema_auth_type === "Basic") {
+      spec.schema_registry.credentials = {
+        username: formData["schema_username"],
+        password: formData["schema_password"],
+      };
+    } else if (formData.schema_auth_type === "API") {
+      spec.schema_registry.credentials = {
+        api_key: formData["schema_api_key"],
+        api_secret: formData["schema_api_secret"],
+      };
+    }
+  }
+
+  return spec;
+}
+
+export function parseTestResult(connection: Connection): PostResponse {
+  let result: { success: boolean; message: string } = { success: false, message: "" };
+
+  const kafkaState: ConnectedState | undefined = connection.status.kafka_cluster?.state;
+  const kafkaErrors: AuthErrors | undefined = connection.status.kafka_cluster?.errors;
+  const schemaRegistryState: ConnectedState | undefined = connection.status.schema_registry?.state;
+  if (kafkaState === "FAILED" || schemaRegistryState === "FAILED") {
```

Comment:
What are your thoughts on mixed `success` states? As an example, this SR config returns `SUCCESS` on its own, but that's not shown when there's a problem with the Kafka config:
<img width="859" alt="image" src="https://github.com/user-attachments/assets/54ed041e-78f5-4da9-8c06-ca3b25e7cde6" />

Reply:
Yes I think that would be smarter - users should see the state of both if they are testing both. Fixing...